sorbet/rbi/shims/rubygems.rbi

```diff
@@ -0,0 +1,34 @@
+# typed: strong
+# frozen_string_literal: true
+
+module Gem
+  class Version
+    sig do
+      params(
+        version: T.nilable(
+          T.any(
+            String,
+            Integer,
+            Gem::Version
+          )
+        )
+      )
+        .returns(Gem::Version)
+    end
```

Comment:
Since we're overriding the underlying signature... do we have valid use cases for integer and nil? Guessing a bunch of stuff breaks if we remove those without first adding types in some places?

Reply:
Integer seems to come from the `docker` updater. Let me see if I can coerce that into a string to simplify this even further.